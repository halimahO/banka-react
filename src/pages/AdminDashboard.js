/* eslint-disable react/prefer-stateless-function */
import React from 'react';

import NavigationBar from '../components/home/NavigationBarStaff';
import Footer from '../components/Footer';

import image from '../img/dashboard-img.png';

class Login extends React.Component {
  render() {
    return (
      <div>
        <NavigationBar />
        <div className="container-dashboard shadow">
          <div className="dashboard">
            <div className="sidebar">
              <div className="head shadow">
                <img id="dashboard-menu-img" src={image} alt="" />
              </div>
              <div className="menu-items menu-body sidebar-li">
                <ul>
                  <li data-menu="profile">
                    <span className="fontAwesame">
                      <i
                        data-menu="profile"
                        className="fas fa-user menu-image"
                      />
                    </span>
                    <span data-menu="profile" className="menu-name">
                      PROFILE
                    </span>
                  </li>
                  <li data-menu="view-an-account">
                    <span className="fontAwesame">
                      <i
                        data-menu="view-an-account"
                        className="far fa-eye menu-image"
                      />
                    </span>
                    <span data-menu="view-an-account" className="menu-name">
                      VIEW ACCOUNT
                    </span>
                  </li>
                  <li data-menu="view-all-accounts">
                    <span className="fontAwesame">
                      <i
                        data-menu="view-all-accounts"
                        className="fas fa-list menu-image"
                      />
                    </span>
                    <span data-menu="view-all-accounts" className="menu-name">
                      ALL ACCOUNTS
                    </span>
                  </li>

                  <li data-menu="change-account-status">
                    <span className="fontAwesame">
                      <i
                        data-menu="change-account-status"
                        className="fas fa-power-off menu-image"
                      />
                    </span>
                    <span
                      data-menu="change-account-status"
                      className="menu-name"
                    >
                      CHANGE STATUS
                    </span>
                  </li>
                  <li data-menu="delete-account">
                    <span className="fontAwesame">
                      <i
                        data-menu="delete-account"
                        className="fas fa-trash-alt menu-image"
                      />
                    </span>
                    <span data-menu="delete-account" className="menu-name">
                      DELETE ACCOUNT
                    </span>
                  </li>
                  <li data-menu="register-a-staff">
                    <span className="fontAwesame">
                      <i
                        data-menu="register-a-staff"
                        className="fas fa-plus-square menu-image"
                      />
                    </span>
                    <span data-menu="register-a-staff" className="menu-name">
                      REGISTER STAFF
                    </span>
                  </li>
                </ul>
              </div>
            </div>
            <div className="display-area">
              <div className="head">
                <p id="menuTopic" className="margin-left">
                  PROFILE
                </p>
              </div>
              <div className="dashboard-area">
                <div id="profile" className="items active">
                  <div className="user-profile">
                    <div className="pix shadow">
                      <span id="profile-pix">
                        <i className="fas fa-user" />
                      </span>
                      <h2 className="m1 md-text">Mercy Ekene</h2>
                      <h4 className="sm-text">mercyekene@gmail.com</h4>
                      <p className="top-border">Admin</p>
                    </div>
                    <div className="profile-details shadow">
                      <div>Firstname: </div>
                      <div>Mercy</div>
                      <div>Lastname:</div>
                      <div>Ekene</div>
                      <div>Email:</div>
                      <div>mercyekene@gmail.com</div>
                      <div>Password:</div>
                      <div>**********</div>
                      <div>.</div>
                      <div>
                        <button type="submit" className="btn btn-reset">
                          Reset password
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  id="view-an-account"
                  className="items dashboard-area-background shadow"
                >
                  <div className="small-acct">
                    <div className="acct-display select-label">
                      <label>Account to view:</label>
                      <br />
                    </div>
                    <div className="acct-display">
                      <input
                        className="dashboard-input acct-number-width"
                        type="text"
                        placeholder="Account number"
                      />
                    </div>
                    <div className="acct-display">
                      <button type="button" className="btn btn-filter cursor">
                        Submit
                      </button>
                    </div>
                  </div>
                  <div className="acct-summary-client shadow">
                    <h4 className="m2">ACCOUNT DETAILS</h4>
                    <p>
                      {' '}
                      <span className="b1">Account Name:</span>
                      Lasisi George
                    </p>
                    <p>
                      <span className="b1">Account Number:</span>
                      0123456789
                    </p>
                    <p>
                      <span className="b1">Opening Date:</span>
                      21/3/2019
                    </p>
                    <p>
                      <span className="b1">Account type:</span>
                      Current
                    </p>
                    <p>
                      <span className="b1">Status:</span>
                      Dormant
                    </p>
                    <p>
                      <span className="b1">Account balance:</span>
                      #2,000.00
                    </p>
                  </div>
                </div>
                <div
                  id="view-all-accounts"
                  className="items dashboard-area-background shadow"
                >
                  <div className="small-acct">
                    <div className="acct-display select-label">
                      <label>Filter by email:</label>
                      <br />
                    </div>
                    <div className="acct-display">
                      <input
                        className="dashboard-input"
                        type="text"
                        placeholder="Email address here..."
                      />
                    </div>
                    <div className="acct-display select-label">
                      <label>Filter by status:</label>
                      <br />
                    </div>
                    <div className="acct-display">
                      <select className="statusSelect">
                        <option value="Savings">None</option>
                        <option value="Savings">Savings</option>
                        <option value="Current">Current</option>
                      </select>
                    </div>
                    <div className="acct-display">
                      {' '}
                      <button type="button" className="btn btn-filter cursor">
                        Submit
                      </button>
                    </div>
                  </div>
                  <div className="acct-summary-client shadow">
                    <h4 className="m2">ALL BANK ACCOUNTS</h4>
                    <div className="table-div">
                      <table>
                        <tbody>
                          <tr>
                            <th>Account Name</th>
                            <th>Owner's Email</th>
                            <th>Account Type</th>
                            <th>Account Balance</th>
                            <th>Account Status</th>
                          </tr>
                          <tr>
                            <td>Halimah Oladosu</td>
                            <td>haleemaholadosu@gmail.com</td>
                            <td>Savings</td>
                            <td>50,000</td>
                            <td>Active</td>
                          </tr>
                          <tr>
                            <td>Halimah Oladosu</td>
                            <td>haleemaholadosu@gmail.com</td>
                            <td>Savings</td>
                            <td>50,000</td>
                            <td>Active</td>
                          </tr>
                          <tr>
                            <td>Halimah Oladosu</td>
                            <td>haleemaholadosu@gmail.com</td>
                            <td>Savings</td>
                            <td>50,000</td>
                            <td>Dormant</td>
                          </tr>
                          <tr>
                            <td>Halimah Oladosu</td>
                            <td>haleemaholadosu@gmail.com</td>
                            <td>Savings</td>
                            <td>50,000</td>
                            <td>Active</td>
                          </tr>
                          <tr>
                            <td>Halimah Oladosu</td>
                            <td>haleemaholadosu@gmail.com</td>
                            <td>Savings</td>
                            <td>50,000</td>
                            <td>Dormant</td>
                          </tr>
                          <tr>
                            <td>Halimah Oladosu</td>
                            <td>haleemaholadosu@gmail.com</td>
                            <td>Savings</td>
                            <td>50,000</td>
                            <td> Active</td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
                <div
                  id="change-account-status"
                  className=" items dashboard-area-background shadow"
                >
                  <div className="small-acct">
                    <div className="acct-display select-label">
                      <label>Change Status:</label>
                      <br />
                    </div>
                    <div className="acct-display">
                      <input
                        className="dashboard-input acct-number-width"
                        type="text"
                        placeholder="Account number"
                      />
                    </div>
                    <div className="acct-display">
                      <button type="button" className="btn btn-filter cursor">
                        Submit
                      </button>
                    </div>
                  </div>
                  <div className="acct-summary-client shadow">
                    <h4 className="m2">ACTIVATE/DEACTIVATE ACCOUNT</h4>
                    <p>
                      <span className="b1">Account Name:</span>
                      Lasisi George
                    </p>
                    <p>
                      <span className="b1">Account Number:</span>
                      0123456789
                    </p>
                    <p>
                      <span className="b1">Opening Date:</span>
                      21/3/2019
                    </p>
                    <p>
                      <span className="b1">Account type:</span>
                      Current
                    </p>
                    <p>
                      <span className="b1">Status:</span>
                      Dormant
                    </p>
                    <p>
                      <span className="b1">Account balance:</span>
                      #2,000.00
                    </p>
                    <button
                      type="button"
                      id="btn-act"
                      className="btn-small cursor"
                      onClick="toggleActiveDeactive()"
                    >
                      ACTIVATE
                    </button>
                  </div>
                </div>
                <div
                  id="delete-account"
                  className=" items dashboard-area-background shadow"
                >
                  <div className="small-acct">
                    <div className="acct-display select-label">
                      <label>Delete Account:</label>
                      <br />
                    </div>
                    <div className="acct-display">
                      <input
                        className="dashboard-input acct-number-width"
                        type="text"
                        placeholder="Account number"
                      />
                    </div>
                    <div className="acct-display">
                      <button type="button" className="btn btn-filter cursor">
                        Submit
                      </button>
                    </div>
                  </div>
                  <div className="acct-summary-client shadow">
                    <h4 className="m2">ACCOUNT DETAILS</h4>
                    <p>
                      <span className="b1">Account Name:</span>
                      Lasisi George
                    </p>
                    <p>
                      <span className="b1">Account Number:</span>
                      0123456789
                    </p>
                    <p>
                      <span className="b1">Opening Date:</span>
                      21/3/2019
                    </p>
                    <p>
                      <span className="b1">Account type:</span>
                      Current
                    </p>
                    <p>
                      <span className="b1">Status:</span>
                      Dormant
                    </p>
                    <p>
                      <span className="b1">Account balance:</span>
                      #2,000.00
                    </p>
                    <button
                      type="button"
                      className="del-btn btn-small btn cursor"
                    >
                      Delete
                    </button>
                  </div>
                </div>
                <div
                  id="register-a-staff"
                  className="items dashboard-area-background content2 shadow"
                >
                  <div>
                    <h1 className="heading-title-login md-title">
                      REGISTER A STAFF/ADMIN
                    </h1>
                  </div>
                  <div className="form-div">
                    <form className="form-div2 shadow">
                      <div>
                        <label>FIRSTNAME</label>

                        <br />
                        <input type="text" placeholder="Firstname" required />
                      </div>
                      <div>
                        <label>LASTNAME</label>

                        <br />
                        <input type="text" placeholder="Lastname" required />
                      </div>
                      <div>
                        <label>EMAIL ADDRESS</label>

                        <br />
                        <input
                          type="email"
                          placeholder="Email Address"
                          required
                        />
                      </div>
                      <div>
                        <label>
                          <p id="password-label">PASSWORD </p>
                        </label>
                        <input type="password" placeholder="*******" required />
                      </div>
                      <div>
                        <label className="container">
                          Is ADMIN?
                          <input type="checkbox" />
                          <span className="checkmark" />
                        </label>
                      </div>
                      <button type="submit" className="btn">
                        <a href="dashboard-staff.html">REGISTER</a>
                      </button>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}
export default Login;
