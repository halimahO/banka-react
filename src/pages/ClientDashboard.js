/* eslint-disable react/jsx-key */
import React from 'react';
import { connect } from 'react-redux';
import NavigationBar from '../components/home/NavigationBarDashboard';
import Footer from '../components/Footer';
import userTransactionAction from '../actions/user-account-action';
import { debit, credit } from '../actions/transaction.action';
import image from '../img/dashboard-img.png';
import PropTypes from 'prop-types';

export class ClientDashboard extends React.Component {
  state = {
    accountNo: '',
    amount: 0
  };

  componentDidMount() {
    const lists = document.querySelectorAll('.menu-items li');
    const removeClass = () => {
      const activeMenu = document.querySelector(
        '.dashboard-area .items.active'
      );
      activeMenu.classList.remove('active');
    };

    const showCurrentMenu = menuId => {
      document.getElementById(menuId).classList.add('active');
    };

    // this.props.fetchUserAccounts(this.props.auth.email);

    Array.from(lists).forEach(listItem => {
      listItem.addEventListener('click', event => {
        const tartgetSectionId = event.target.dataset.menu;
        removeClass();
        showCurrentMenu(tartgetSectionId);
        document.getElementById(
          'menuTopic'
        ).innerHTML = tartgetSectionId.replace(/-/g, ' ');
      });
    });
  }

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  handleDebit = async e => {
    e.preventDefault();
    const amount = { amount: parseFloat(this.state.amount) };
    await this.props.debit(this.state.accountNo, amount);
  };

  handleCredit = async e => {
    e.preventDefault();
    const amount = { amount: parseFloat(this.state.amount) };
    await this.props.credit(this.state.accountNo, amount);
  };

  handleTransactions = async e => {
    e.preventDefault();
    await this.props.UserTransaction(this.state.accountNo);
  };

  render() {
    return (
      <div>
        <NavigationBar />
        <div className="container-dashboard shadow">
          <div className="dashboard">
            <div className="sidebar">
              <div className="head shadow">
                <img id="dashboard-menu-img" src={image} alt="" />
              </div>
              <div className="menu-items menu-body sidebar-li">
                <ul>
                  <li data-menu="profile">
                    <span className="fontAwesame">
                      <i
                        data-menu="profile"
                        className="fas fa-user menu-image"
                      />
                    </span>
                    <span data-menu="profile" className="menu-name">
                      PROFILE
                    </span>
                  </li>
                  <li data-menu="account-history">
                    <span className="fontAwesame">
                      <i
                        data-menu="account-history"
                        className="fas fa-file menu-image"
                      />
                    </span>

                    <span data-menu="account-history" className="menu-name">
                      ACCOUNT HISTORY
                    </span>
                  </li>
                  {this.props.auth.type !== 'client' ? (
                    <React.Fragment>
                      <li data-menu="debit-account">
                        <span className="fontAwesame">
                          <i
                            data-menu="debit-account"
                            className="fas fa-credit-card menu-image"
                          />
                        </span>
                        <span data-menu="debit-account" className="menu-name">
                          DEBIT ACCOUNT
                        </span>
                      </li>
                      <li data-menu="credit-account">
                        <span className="fontAwesame">
                          <i
                            data-menu="credit-account"
                            className="fas fa-credit-card menu-image"
                          />
                        </span>
                        <span data-menu="credit-account" className="menu-name">
                          CREDIT ACCOUNT
                        </span>
                      </li>
                    </React.Fragment>
                  ) : null}
                </ul>
              </div>
            </div>
            <div className="display-area">
              <div className="head">
                <p id="menuTopic" className="margin-left">
                  PROFILE
                </p>
              </div>
              <div className="dashboard-area">
                <div id="profile" className="items active">
                  <div className="user-profile">
                    <div className="pix shadow">
                      <span id="profile-pix">
                        <i className="fas fa-user" />
                      </span>
                      <h2 className="m1 md-text">{`${this.props.auth.firstname} ${this.props.auth.lastname}`}</h2>
                      <h4 className="sm-text">{this.props.auth.email}</h4>
                      <p className="top-border">{this.props.auth.type}</p>
                    </div>
                    <div className="profile-details shadow">
                      <div>Firstname:</div>
                      <div>{this.props.auth.firstname}</div>
                      <div>Lastname:</div>
                      <div>{this.props.auth.lastname}</div>
                      <div>Email:</div>
                      <div>{this.props.auth.email}</div>
                      <div>Password:</div>
                      <div>**********</div>
                      <div>Account Number:</div>
                      <div>
                        {Object.keys(this.props.newAccount).length > 0
                          ? this.props.newAccount.account.accountNumber
                          : ''}
                      </div>
                      <div>.</div>
                      <div>
                        <button type="submit" className="btn btn-reset">
                          Reset password
                        </button>
                      </div>
                    </div>
                  </div>
                </div>

                <div
                  id="account-history"
                  className="items dashboard-area-background shadow"
                >
                  <div>
                    <div className="small-acct">
                      <div className="acct-display select-label">
                        <label htmlFor="accountNo">Account to display:</label>
                        <br />
                      </div>
                      <div className="acct-display">
                        <input
                          className="dashboard-input acct-number-width"
                          name="accountNo"
                          type="number"
                          id="accountNo"
                          onChange={this.handleChange}
                          placeholder="Account number here"
                        />
                      </div>
                      <div className="acct-display">
                        <button
                          type="submit"
                          className="btn btn-filter cursor"
                          onClick={this.handleTransactions}
                        >
                          Submit
                        </button>
                      </div>
                    </div>
                  </div>
                  <h4 className="m2 clear">
                    Account:
                    {this.state.accountNo}
                  </h4>
                  <div className="table-div">
                    <table>
                      <thead>
                        <tr>
                          <th>Transaction Id</th>
                          <th>Transaction Date</th>
                          <th>Transaction type</th>
                          <th>Amount</th>
                          <th>Old Balance</th>
                          <th>New Balance</th>
                        </tr>
                      </thead>
                      <tbody>
                        {this.props.userTransactions.map(transaction => {
                          return (
                            <tr key={transaction.id}>
                              <td>{transaction.id}</td>
                              <td>{transaction.createdon}</td>
                              <td>{transaction.type}</td>
                              <td>{transaction.amount}</td>
                              <td>{transaction.oldbalance}</td>
                              <td>{transaction.newbalance}</td>
                            </tr>
                          );
                        })}
                      </tbody>
                    </table>
                  </div>
                </div>
                <div
                  id="debit-account"
                  className="items dashboard-area-background shadow"
                >
                  <div className="small-acct">
                    <div className="acct-display select-label">
                      <label htmlFor="accountNo">Debit Account:</label>
                      <br />
                    </div>
                    <div className="acct-display">
                      <input
                        className="dashboard-input acct-number-width"
                        name="accountNo"
                        type="number"
                        onChange={this.handleChange}
                        placeholder="Account number"
                      />
                    </div>
                  </div>
                  <div className="acct-summary-client shadow">
                    <h4 className="m2">DEBIT THE ACCOUNT</h4>
                    <input
                      className="amountInput"
                      type="number"
                      name="amount"
                      onChange={this.handleChange}
                      placeholder="Amount..."
                    />
                    <button
                      id="handleDebit"
                      type="submit"
                      className="btn-small cursor"
                      onClick={this.handleDebit}
                    >
                      DEBIT
                    </button>
                  </div>
                </div>
                <div
                  id="credit-account"
                  className="items dashboard-area-background shadow"
                >
                  <div className="small-acct">
                    <div className="acct-display select-label">
                      <label htmlFor="accountNo">Credit Account:</label>
                      <br />
                    </div>
                    <div className="acct-display">
                      <input
                        className="dashboard-input acct-number-width"
                        type="text"
                        name="accountNo"
                        onChange={this.handleChange}
                        placeholder="Account number"
                      />
                    </div>
                  </div>
                  <div className="acct-summary-client shadow">
                    <h4 className="m2">CREDIT THE ACCOUNT</h4>
                    <input
                      className="amountInput"
                      type="number"
                      name="amount"
                      onChange={this.handleChange}
                      placeholder="Amount..."
                    />
                    <button
                      type="submit"
                      className="btn-small cursor"
                      onClick={this.handleCredit}
                    >
                      CREDIT
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

ClientDashboard.propTypes = {
  auth: PropTypes.object,
  fetchUserAccounts: PropTypes.func,
  UserTransaction: PropTypes.func,
  debit: PropTypes.func,
  credit: PropTypes.func,
  userTransactions: PropTypes.any,
  accountNumber: PropTypes.number,
  newAccount: PropTypes.object
};
export const mapStateToProps = state => {
  return {
    auth: state.auth.user,
    userTransactions: state.userTransactions,
    newAccount: state.newAccount
  };
};

export const mapDispatchToProps = dispatch => {
  return {
    UserTransaction: accountNo => dispatch(userTransactionAction(accountNo)),
    debit: (accountNo, amount) => dispatch(debit(accountNo, amount)),
    credit: (accountNo, amount) => dispatch(credit(accountNo, amount))
    // fetchUserAccounts: emailAddress => dispatch(fetchUserAccounts(emailAddress))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ClientDashboard);
