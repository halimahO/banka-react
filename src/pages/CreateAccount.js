import React from 'react';
import Select from 'react-select';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';
import NavigationBar from '../components/home/NavigationBarDashboard';
import Footer from '../components/Footer';
import createAccountAction from '../actions/create-account.action';

const options = [
  { value: 'savings', label: 'Savings' },
  { value: 'current', label: 'Current' }
];
export class CreateAccountPage extends React.Component {
  state = {
    selectedAccountType: null
  };

  handleChange = selectedAccountType => {
    this.setState({ selectedAccountType });
  };

  handleSubmit = async e => {
    const { history } = this.props;
    e.preventDefault();
    try {
      const type = { type: this.state.selectedAccountType.value };
      await this.props.createAccountAction(type, history);
    } catch (error) {
      return toast.error('Account type is required');
    }
  };

  render() {
    const { selectedAccountType } = this.state;

    return (
      <div>
        <NavigationBar />
        <div id="login" className="padding-bot border shadow">
          <div>
            <h1 className="heading-title-login m2">OPEN AN ACCOUNT</h1>
          </div>
          <div className="form-div">
            <form onSubmit={this.handleSubmit}>
              <div>
                <label htmlFor="fullname">FULLNAME</label>
                <p>{`${this.props.firstname} ${this.props.lastname}`}</p>
              </div>
              <div>
                <label htmlFor="SelectedAccountType">ACCOUNT TYPE</label>
                <br />
                <br />
                <Select
                  className="selectChange"
                  value={selectedAccountType}
                  onChange={this.handleChange}
                  options={options}
                />
              </div>
              <button type="submit" className="btn">
                OPEN ACCOUNT
              </button>
            </form>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}
CreateAccountPage.propTypes = {
  history: PropTypes.object.isRequired,
  createAccountAction: PropTypes.func,
  firstname: PropTypes.string,
  lastname: PropTypes.string
};

export const mapStateToProps = state => {
  return {
    firstname: state.auth.user.firstname,
    lastname: state.auth.user.lastname
  };
};
export const mapDispatchToProps = dispatch => {
  return {
    createAccountAction: async (userCredentials, history) => {
      return dispatch(await createAccountAction(userCredentials, history));
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CreateAccountPage);
