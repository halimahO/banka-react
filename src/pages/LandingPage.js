/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import NavigationBar from '../components/home/NavigationBar';

class LandingPage extends React.Component {
  render() {
    return (
      <div>
        <div>
          <div>AppName</div>
          <div>
            <NavigationBar />
          </div>
        </div>
        <div>
          <div>AppName</div>
          <div>AppDesciption</div>
          <div>getStarted</div>
        </div>
        <div>
          <div>oursevices</div>
          <div>images</div>
        </div>
        <div>
          <div>social text</div>
          <div>social</div>
          <div>copyright</div>
        </div>
      </div>
    );
  }
}
export default LandingPage;
