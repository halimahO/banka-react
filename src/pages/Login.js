/* eslint-disable react/prefer-stateless-function */
import React from 'react';

import NavigationBar from '../components/home/NavigationBarLogin';
import Footer from '../components/Footer';

class Login extends React.Component {
  render() {
    return (
      <div>
        <NavigationBar />
        <div className="login-section padding-bot border shadow">
          <div>
            <h1 className="heading-title-login">SIGN IN</h1>
            <p className="center">
              Not registered with us yet?
              <a href="register.html">Sign up</a>
            </p>
          </div>
          <div className="form-div">
            <form onSubmit="return goToDashboard()">
              <div>
                <label>EMAIL ADDRESS</label>
                <br />
                <input
                  id="emailInput"
                  type="email"
                  placeholder="Email Address"
                  required
                />
              </div>
              <div>
                <label>
                  {' '}
                  <p id="password-label">PASSWORD </p>
                </label>
                <input type="password" placeholder="*******" required />
                <a href>
                  <p id="forgot-password-ptag">Forgot your password?</p>
                </a>
              </div>
              <button type="submit" id="loginButton" className="btn">
                SIGN IN
              </button>
            </form>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}
export default Login;
