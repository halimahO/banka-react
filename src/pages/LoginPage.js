import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import NavigationBar from '../components/home/NavigationBarLogin';
import Footer from '../components/Footer';
import loginAction from '../actions/login.action';

class LoginPage extends React.Component {
  state = {
    email: '',
    password: ''
  };

  handleChange = e => {
    this.setState({
      [e.target.name]: e.target.value
    });
  };

  handleSubmit = async e => {
    const { history } = this.props;
    e.preventDefault();
    await this.props.loginAction(this.state, history);
  };

  render() {
    return (
      <div>
        <NavigationBar />
        <div className="login-section padding-bot border shadow">
          <div>
            <h1 className="heading-title-login">SIGN IN</h1>
            <p className="center">
              Not registered with us yet?
              <a href="register.html">Sign up</a>
            </p>
          </div>
          <div className="form-div">
            <form onSubmit={this.handleSubmit}>
              <div>
                <label htmlFor="emailInput">EMAIL ADDRESS</label>
                <br />
                <input
                  id="emailInput"
                  type="email"
                  name="email"
                  placeholder="Email Address"
                  onChange={this.handleChange}
                  required
                />
              </div>
              <div>
                <label htmlFor="password">
                  <p id="password-label">PASSWORD </p>
                </label>
                <input
                  type="password"
                  name="password"
                  placeholder="*******"
                  required
                  onChange={this.handleChange}
                />
                <a href="#1">
                  <p id="forgot-password-ptag">Forgot your password?</p>
                </a>
              </div>
              <button type="submit" id="loginButton" className="btn">
                SIGN IN
              </button>
            </form>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

LoginPage.propTypes = {
  history: PropTypes.object.isRequired,
  loginAction: PropTypes.func.isRequired
};

const mapDispatchToProps = dispatch => {
  return {
    loginAction: async (userCredentials, history) => {
      return dispatch(await loginAction(userCredentials, history));
    }
  };
};

export default connect(
  null,
  mapDispatchToProps
)(LoginPage);
