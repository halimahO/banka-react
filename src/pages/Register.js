/* eslint-disable react/prefer-stateless-function */
import React from 'react';

import NavigationBarShadow from '../components/home/NavigationBarShadow';
import Footer from '../components/Footer';

class Register extends React.Component {
  render() {
    return (
      <div>
        <NavigationBarShadow />
        <div className="login-section padding-bot border shadow">
          <div>
            <h1 className="heading-title-login">REGISTER</h1>
            <p className="center">
              Already have an account?
              <a href="sign-in.html">Sign in</a>
            </p>
          </div>
          <div className="form-div">
            <form>
              <div>
                <label>FIRSTNAME</label>
                <br />
                <input type="text" placeholder="Firstname" required />
              </div>
              <div>
                <label>LASTNAME</label>
                <br />
                <input type="text" placeholder="Lastname" required />
              </div>
              <div>
                <label>EMAIL ADDRESS</label>
                <br />
                <input type="email" placeholder="Email Address" required />
              </div>
              <div>
                <label>
                  <p id="password-label">PASSWORD </p>
                </label>
                <input type="password" placeholder="*******" required />
              </div>
              <button type="submit" location="dashboard.html" className="btn">
                REGISTER
              </button>
            </form>
          </div>
        </div>

        <Footer />
      </div>
    );
  }
}
export default Register;
