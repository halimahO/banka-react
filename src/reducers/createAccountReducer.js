import * as types from '../action-types/index';

const createAccountReducer = (state = {}, action) => {
  switch (action.type) {
    case types.CREATE_ACCOUNT_ERROR:
      return { ...state, error: action.payload };
    case types.CREATE_ACCOUNT_SUCCESS:
      return { ...state, account: action.payload };
    default:
      return state;
  }
};

export default createAccountReducer;
