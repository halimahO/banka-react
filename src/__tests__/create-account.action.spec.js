import axios from 'axios';
import thunk from 'redux-thunk';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import { toast } from 'react-toastify';
import { CreateAccountPage, mapDispatchToProps } from '../pages/CreateAccount';
import { createAccountAction } from '../actions/create-account.action';
import {
  CREATE_ACCOUNT_SUCCESS,
  CREATE_ACCOUNT_ERROR
} from '../action-types/index';
import createAccountReducer from '../reducers/createAccountReducer';

jest.mock('axios');

describe('Create Account Page component Tests', () => {
  const props = {
    createAccountAction: jest.fn(),
    history: {},
    firstname: 'name',
    lastname: 'lastname'
  };

  it('renders the CreateAccount component correctly', () => {
    const wrapper = shallow(<CreateAccountPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should simulate an onchange event on form input', () => {
    const event = {
      preventDefault() {},
      target: { name: 'leemar', value: 'leemar@mail.com' }
    };
    const component = mount(
      <BrowserRouter>
        <CreateAccountPage {...props} />
      </BrowserRouter>
    );

    const inputTag = component.find('.selectChange').at(0);
    inputTag.simulate('change', event);
  });

  it('should render component successfully and check form interactions', () => {
    const component = mount(
      <BrowserRouter>
        <CreateAccountPage {...props} />
      </BrowserRouter>
    );
    component.find('form').simulate('submit');
  });

  it('should dispatch createAccountAction request action', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).createAccountAction();
  });
});
// describe('Create Account Actions', () => {
//   const type = {
//     type: 'savings'
//   };
//   const middleware = [thunk];
//   const mockStore = configureMockStore(middleware);
//   let store;
//   beforeEach(() => {
//     store = mockStore({});
//     jest.resetAllMocks();
//   });
//   afterEach(() => {
//     store.clearActions();
//   });

//   it('Should Trigger the CREATE_ACCOUNT dispatch function', async () => {
//     const mockData = {
//       status: 201,
//       data: {
//         data: {
//           accountNumber: 2314963840,
//           firstName: 'Alde',
//           lastName: 'Sode',
//           email: 'adeubade@mail.com',
//           type: 'savings',
//           openingBalance: 0
//         }
//       }
//     };

//     axios.post.mockResolvedValueOnce(mockData);

//     const expectedActions = [
//       { type: 'CREATE_ACCOUNT_SUCCESS', payload: mockData }
//     ];

//     store.dispatch(createAccountAction(type)).then(() => {
//       expect(store.getActions()).toEqual(expectedActions);
//     });
//   });

//   it('Should Trigger the CREATE_ACCOUNT_ERROR dispatch function', async () => {
//     const mockData = {
//       response: {
//         data: {
//           status: 400,
//           error: 'An error occured'
//         }
//       }
//     };

//     axios.post.mockRejectedValueOnce(mockData);

//     const expectedActions = [
//       { type: CREATE_ACCOUNT_ERROR, payload: mockData.response.data.error }
//     ];
//     store.dispatch(createAccountAction(type)).then(() => {
//       expect(store.getActions()).toEqual(expectedActions);
//     });
//   });

//   it('Should return a toast on network error', async () => {
//     const mockData = {
//       message: 'Network Error'
//     };

//     const spy = jest.spyOn(toast, 'error');

//     axios.post.mockRejectedValueOnce(mockData);

//     store.dispatch(createAccountAction(type)).then(() => {
//       expect(spy).toHaveBeenCalledWith('An error occured. Please try Again');
//     });
//   });
// });

// describe('Create Account Reducer', () => {
//   it('Should return a new state if it recieves a create account success action type', () => {
//     const account = {
//       accountNumber: 2579447706,
//       firstName: 'Ade',
//       lastName: 'Bade',
//       email: 'adebade@gmail.com',
//       type: 'current',
//       openingBalance: 0
//     };
//     const newState = createAccountReducer(undefined, {
//       type: CREATE_ACCOUNT_SUCCESS,
//       payload: account
//     });
//     expect(newState).toEqual({
//       account
//     });
//   });

//   it('Should return a new state if it recieves create account action type', () => {
//     const message = 'invalid account number/ amount';
//     const newState = createAccountReducer(undefined, {
//       type: CREATE_ACCOUNT_ERROR,
//       payload: message
//     });
//     expect(newState).toEqual({
//       error: message
//     });
//   });
// });
