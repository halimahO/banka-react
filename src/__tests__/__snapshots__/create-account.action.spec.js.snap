// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Account Page component Tests renders the CreateAccount component correctly 1`] = `ShallowWrapper {}`;
